docs/book/src/reference/creating-events.md

```diff
@@ -0,0 +1,109 @@
+# Creating Events
+
+It is often useful to publish *Event* objects from the controller Reconcile function. 
+
+Events allow users to see what is going on with a particular object, and allow automated processes to see and respond to them.
+
+ Recent Events for an object can be viewed by running `kubectl describe <resource kind> <resource name>`.
+ 
+  ### how to implement the solution to raise an event:
+
+a) You need to pass the recorder when you set up the recorder for the controller when the event will be raised see: https://github.com/kubernetes-sigs/kubebuilder/blob/master/testdata/project-v4-with-deploy-image/main.go#L103
+b) You need to add the makers to add the RBAC permissions to allow your Operator/controller to raise the event see: https://github.com/kubernetes-sigs/kubebuilder/blob/master/testdata/project-v4-with-deploy-image/controllers/memcached_controller.go#L66 and run make manifests
+c) You can check an example of the event being called in: https://github.com/kubernetes-sigs/kubebuilder/blob/master/testdata/project-v4-with-deploy-image/controllers/memcached_controller.go#L299-L303
```

Comment:
````suggestion
### Granting the required permissions

You must also grant the RBAC rules permissions to allow your project to create Events. Therefore, ensure that you add the [RBAC][rbac-markers] into your controller:

```go
...
//+kubebuilder:rbac:groups=core,resources=events,verbs=create;patch
...

func (r *MyKindReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
```

And then, run `$ make manifests` to update the rules under `config/rbac/rule.yaml`. 
````

Reply:
We will need to add at the bottom an alias `[rbac-markers]` for the page: https://book.kubebuilder.io/reference/markers/rbac.html

It is a kubebuilder doc, so we should NOT use the above URL; instead, we should use a relative path from this place to the file.

> [rbac-markers]: markers/rbac.md